tests/unit/plugins/kubernetes/kubernetes_pod_executor_test.py

```diff
@@ -318,6 +318,55 @@ def test_pending_event_processing_loop_processes_remaining_events_after_stop(k8s
     assert k8s_executor.pending_events.qsize() == 0
 
 
+@mock.patch(
+    "task_processing.plugins.kubernetes.kubernetes_pod_executor.REFRESH_EXECUTOR_STATE_THREAD_GRACE",  # noqa
+    1,
+)
+@mock.patch(
+    "task_processing.plugins.kubernetes.kubernetes_pod_executor.REFRESH_EXECUTOR_STATE_THREAD_INTERVAL",  # noqa
+    1,
+)
+def test_reconcile_task_loop_reconcile_existing_pods_only(k8s_executor, mock_task_configs):
+    mock_pods = []
+    test_phases = ['Unknown', 'Succeeded', 'Failed', 'Running']
+    task_metadata = {}
+
+    for taskconf, phase in zip(mock_task_configs, test_phases):
+        mock_pod = mock.Mock(spec=V1Pod)
+        mock_pod.metadata.name = taskconf.pod_name
+        mock_pod.status.phase = phase
+        mock_pod.status.host_ip = '1.2.3.4'
+        mock_pod.spec.node_name = 'kubenode'
+        mock_pods.append(mock_pod)
+        task_metadata[taskconf.pod_name] = KubernetesTaskMetadata(
+            task_config=taskconf,
+            task_state=KubernetesTaskState.TASK_RUNNING,
+            task_state_history=v(),
+        )
+    # Keep only 3 task_configs in task_metadata
+    task_metadata.popitem()
+    k8s_executor.task_metadata = pmap(task_metadata)
+
+    with mock.patch.object(
+        k8s_executor,
+        "reconcile",
+        autospec=True,
+    ) as mock_reconcile_task, mock.patch.object(
+        k8s_executor,
+        "kube_client",
+        autospec=True
+    ) as mock_kube_client, mock.patch(
+        "task_processing.plugins.kubernetes.kubernetes_pod_executor.KubernetesPodExecutor.is_stopping",  # noqa
+        new_callable=mock.PropertyMock,
+        side_effect=[False, True]
+    ):
+        mock_kube_client.get_pods.return_value = mock_pods
+        k8s_executor._reconcile_task_loop()
+
+    mock_reconcile_task.assert_called()
+    assert mock_reconcile_task.call_count == 3
```

Comment:
should we add an assert that the reconcile loop either changed/didn't change anything?

Reply:
the reconcile function here is actually mocked so it won't actually modify the events. This test here is simply to ensure that we are calling the reconcile function the right number of times based on what we see mock_pods/task_metadata. 

The tests for actually reconciling is handled separately in other tests.